resolving/npm-resolver/src/index.ts

```diff
@@ -251,9 +252,15 @@ async function resolveNpm (
     resolution,
     resolvedVia: 'npm-registry',
     publishedAt: meta.time?.[pickedPackage.version],
+    specifierTemplate: opts.calcSpecifierTemplate ? calcSpecifierTemplate(wantedDependency.alias, spec) : undefined,
   }
 }
 
+function calcSpecifierTemplate (alias: string | undefined, spec: RegistryPackageSpec): string {
+  if (!alias || spec.name === alias) return '<range>'
+  return `npm:${spec.name}@<range>`
+}
```

Comment:
I feel like having the template being a string would be suboptimal:
* It forces parsing.
* Parsing is also error-prone.

Couldn't it be an object instead, e.g.

```typescript
interface SpecifierTemplate {
  toString: () => string
}
```

or:

```typescript
abstract class SpecifierTemplate {
  abstract toString (): string
}
```

Reply:
The template is removed. The specifier calculation together with the range calculation is moved to the resolver.